src/settings.rs

```diff
@@ -105,9 +109,47 @@ impl kubewarden::settings::Validatable for Settings {
             return Err("Signatures must not be empty".to_string());
         }
 
-        // TODO: when a certificate is being used, ensure the certificate
-        // can be trusted. This requires a new waPC function being written
-
+        // when a certificate is being used, ensure it can be trusted
+        for signature in self.signatures.iter() {
+            if let Signature::Certificate(s) = signature {
+                // build sdk structs:
+                let cert = SDKCert {
+                    encoding: CertificateEncoding::Pem,
+                    data: s.certificate.clone().into_bytes(),
+                };
+                let cert_chain_opt: Option<Vec<SDKCert>> = match &s.certificate_chain {
```

Comment:
nitpicking: the `match` could be rewritten using a `map`. I think something like:

```rust
let cert_chain_opt: Option<Vec<SDKCert>> = &.s.certificate_chain.map({ |chain|
   chain.iter().map(|c| SDKCert{
                                encoding: CertificateEncoding::Pem,
                                data: c.clone().into_bytes(),
   }).collect()
});
```

Maybe there's also a way to remove the `clone()`

Reply:
that's way better, didn't think about a `map`.

Done as part of https://github.com/kubewarden/verify-image-signatures/pull/47/commits/dd0f8852b0367166405b1f1b919bf525770075ca, but couldn't reduce the clone (now it's a `to_owned()`).